ops/model.py

```diff
@@ -111,6 +112,11 @@ def name(self) -> str:
         """
         return self._backend.model_name
 
+    @property
+    def goal(self) -> 'GoalState':
+        """Return :class: GoalState object representing the goal-state of the Juju model."""
+        return self._goal
```

Comment:
So I don't think we want to expose this as Goal State, since we have a plan to deprecate 'goal-state' as the way to get this sort of information.

Instead, we should be exposing expected_units / desired_units as part of other parts of the model (such as on Relation objects).
It can use goal-state for how it determines that information, but we don't want to expose charmers to `self.model.goal.STUFF`